lib/osf-components/addon/components/search-result-card/component.ts

```diff
@@ -1,31 +1,81 @@
 import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Store from '@ember-data/store';
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
-import SearchResultModel from 'ember-osf-web/models/search-result';
-import { inject as service } from '@ember/service';
 import Intl from 'ember-intl/services/intl';
+import { dropTask } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+
+import config from 'ember-osf-web/config/environment';
+import SearchResultModel from 'ember-osf-web/models/search-result';
+import UserModel from 'ember-osf-web/models/user';
+
+const osfUrl = config.OSF.url;
 
 interface Args {
     result: SearchResultModel;
 }
 
 export default class SearchResultCard extends Component<Args> {
     @service intl!: Intl;
+    @service store!: Store;
+
     @tracked isOpenSecondaryMetadata = false;
+    @tracked osfUser?: UserModel;
 
     @action
     toggleSecondaryMetadata() {
         this.isOpenSecondaryMetadata = !this.isOpenSecondaryMetadata;
+        if (this.isUserResultCard && !this.osfUser) {
+            taskFor(this.getOsfUserModel).perform();
+        }
     }
 
     get cardTypeLabel() {
         return this.intl.t(`osf-components.search-result-card.${this.args.result.resourceType}`);
     }
 
-    // not sure if this is the best way, as there was a resourceType of "unknown" out in the wild
+    get isUserResultCard() {
+        return this.args.result.resourceType === 'user';
+    }
+
     get secondaryMetadataComponent() {
         const { resourceType } = this.args.result;
 
         return `search-result-card/${resourceType.replace('_component', '')}-secondary-metadata`;
```

Comment:
(sorry for stepping outside the diff -- consider it a nitpick) would recommend an explicit map from `resourceType` to complete component name, so component names are not entangled with data and so you can find where components are used by searching for them